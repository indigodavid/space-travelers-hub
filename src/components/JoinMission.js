import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { joinMission, leaveMission } from '../redux/missions/missions';

const JoinMission = (props) => {
  const { reserved, id } = props;
  const dispatch = useDispatch();
  const handleJoin = () => {
    dispatch(joinMission(id));
  };

  const handleLeave = () => {
    dispatch(leaveMission(id));
  };

  if (reserved) {
    return (
      <button type="button" className="leave" onClick={handleLeave}>Leave Mission</button>
    );
  }
  return (
    <button type="button" className="join" onClick={handleJoin}>Join Mission</button>
  );
};

JoinMission.propTypes = {
  reserved: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
};

export default JoinMission;
