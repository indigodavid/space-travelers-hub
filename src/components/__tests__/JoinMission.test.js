import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore'
import JoinMission from '../JoinMission';


const renderJoinMission = () => {
  const { container } = render(
    <Provider store={store}>
      <JoinMission reserved={true} id="9D1B7E0" />
    </Provider>
  );
  return container;
}

describe('Test JoinMission component', () => {
  test('Match Snapshot', () => {
    expect(renderJoinMission().firstChild).toMatchSnapshot();
  })

  test('Test button con click', () => {
    const element = renderJoinMission()
    expect(element.querySelector('.leave')).toBeInTheDocument();
    fireEvent.click(screen.getByText(/Leave Mission/i));
    setTimeout(() => {
      expect(element.querySelector('.join')).toBeInTheDocument();
    }, 500);
  })
})

