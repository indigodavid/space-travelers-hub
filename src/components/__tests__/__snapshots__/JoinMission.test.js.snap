// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test JoinMission component Match Snapshot 1`] = `
<button
  class="leave"
  type="button"
>
  Leave Mission
</button>
`;
